pkg/webhooks/customresourcedefinitions/customresourcedefinitions.go

```diff
@@ -0,0 +1,201 @@
+package customresourcedefinitions
+
+import (
+	"fmt"
+	"net/http"
+	"regexp"
+	"slices"
+	"strings"
+
+	"github.com/openshift/managed-cluster-validating-webhooks/pkg/webhooks/utils"
+
+	admissionregv1 "k8s.io/api/admissionregistration/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	admissionctl "sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+const (
+	WebhookName string = "customresourcedefinitions-validation"
+	docString   string = `Managed OpenShift Customers may not change CustomResourceDefinitions managed by Red Hat.`
+)
+
+var (
+	timeout                          int32 = 2
+	allowedUsers                           = []string{"system:admin", "backplane-cluster-admin"}
+	sreAdminGroups                         = []string{"system:serviceaccounts:openshift-backplane-srep"}
+	privilegedServiceAccountGroupsRe       = regexp.MustCompile(utils.PrivilegedServiceAccountGroups)
+	scope                                  = admissionregv1.ClusterScope
+	rules                                  = []admissionregv1.RuleWithOperations{
+		{
+			Operations: []admissionregv1.OperationType{
+				admissionregv1.Create,
+				admissionregv1.Update,
+				admissionregv1.Delete,
+			},
+			Rule: admissionregv1.Rule{
+				APIGroups:   []string{"apiextensions.k8s.io"},
+				APIVersions: []string{"*"},
+				Resources:   []string{"customresourcedefinitions"},
+				Scope:       &scope,
+			},
+		},
+	}
+	log = logf.Log.WithName(WebhookName)
+)
+
+// customresourcedefinitionsruleWebhook validates a customresourcedefinition change
+type customresourcedefinitionsruleWebhook struct {
+	s runtime.Scheme
+}
+
+// We just need a runtime object to get the labels
+type customResourceDefinition struct {
+	runtime.Object
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+}
```

Comment:
We shouldn't be defining our own type here and instead use the pre-defined type upstream within apiextensions, e.g. https://github.com/openshift/managed-cluster-validating-webhooks/blob/master/pkg/webhooks/namespace/namespace.go#L158

Reply:
Here's the CRD type upstream https://github.com/kubernetes/apiextensions-apiserver/blob/master/pkg/apis/apiextensions/v1/types.go#L367C24-L379